dra-evolution/pkg/api/claim_types.go

```diff
@@ -376,6 +433,14 @@ type DeviceConfiguration struct {
 	DriverConfigurationParameters `json:",inline"`
 }
 
+// DriverConfigurationParameters must have one and only one one field set.
+//
+// In contrast to ConfigurationParameters, the driver name is
+// not included and has to be infered from the context.
+type DriverConfigurationParameters struct {
+	Opaque *runtime.RawExtension `json:"opaque,omitempty" protobuf:"bytes,1,opt,name=opaque"`
+}
+
```

Comment:
I don't quite understand this comment. The `ConfigurationParameters` and `DriverConfigurationParameters` structs look identical to me.

Reply:
Edit: I see the difference now -- the other struct has an embedded `OpaqueConfigurationParameters` with an embedded `DriveName` field. rather than a direct field for `runtime.RawExtension`.

It's still confusing to look at, but I at least see the difference now...

Reply:
Would it make more sense to repeat the driver name, i.e. use `ConfigurationParameters`?

That could look odd in YAML because the driver name will be more obviously redundant:
```yaml
results:
- driverName: dra.example.com
  config:
  - opaque:
      driverName: dra.example.com
      parameters:
        ...
```

Reply:
Perhaps it actually makes sense? One driver could understand the parameters of another.

We don't do that with structured parameters (it's defined as "only configs for the driver are passed down to it"), but perhaps that could change and a control plane controller could already do it now.

Reply:
Not added yet to https://github.com/kubernetes/enhancements/pull/4709 but perhaps we should?

If we do, we probably should make it consistent and also pass down all configurations down when using structured parameters.